core/src/toga/app.py

```diff
@@ -520,11 +520,60 @@ def current_window(self):
         """Return the currently active content window."""
         return self._impl.current_window().interface
 
+    @property
+    def maximized(self):
+        """Returns True if any of the app's windows are maximized."""
+        return any(window.maximized for window in self.windows)
+
+    @maximized.setter
+    def maximized(self, maximize: bool):
+        if maximize:
+            for window in self.windows:
+                window.maximized = True
+        else:
+            for window in self.windows:
+                window.maximized = False
+
+    @property
+    def minimized(self):
+        """Returns True only if all of the app's windows are minimized."""
+        return all(window.minimized for window in self.windows)
+
+    @minimized.setter
+    def minimized(self, minimize: bool):
+        if minimize:
+            for window in self.windows:
+                window.minimized = True
+        else:
+            for window in self.windows:
+                window.minimized = False
+
+    @property
+    def full_screen(self):
+        """Return True if any of the app's window are in full screen."""
+        return any(window.full_screen for window in self.windows)
+
+    @full_screen.setter
+    def full_screen(self, full_screen):
+        if full_screen:
+            for window in self.windows:
+                window.full_screen = True
+        else:
+            for window in self.windows:
+                window.full_screen = False
+
+    def set_normal_screen(self):
+        """Sets the app's windows in normal state"""
+        for window in self.windows:
+            window.set_normal_screen()
+
+    # For backwards compatibility in core API
     @property
     def is_full_screen(self):
         """Is the app currently in full screen mode?"""
         return self._full_screen_windows is not None
 
+    # For backwards compatibility in core API
```

Comment:
This isn't backwards-compatibility - it's maintaining a feature set: the ability to set a *selection* of windows full screen.

Reply:
I will modify it to add a DeprecationWarning and change the internal implementation.